File: 0053-最大子序和/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        if(nums.length == 0){
            return 0;
        }
        int max = nums[0];
        for(int i = 0; i<nums.length; i++){
            int sum = 0;
            for(int j = i; j<nums.length; j++){
                sum += nums[j];
                if(sum > max){
                    max = sum;
                }
            }
        }
        return max;
        
        
    }
}

File: 0053-最大子序和/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = nums[0]
        prev = nums[0]
        for i in range(1, len(nums)):
            curr = max(nums[i], nums[i]+prev)
            prev = curr
            res = max(res, curr)
        return res
